src/contract_client/client.ts

```diff
@@ -45,6 +47,35 @@ export class ContractClient {
     });
   }
 
+  /**
+   * Generate a ContractClient instance from the ContractClientOptions and the wasm binary
+   */
+  static async fromWasm(options: ContractClientOptions, wasm: BufferSource): Promise<ContractClient> {
+    const wasmModule = await WebAssembly.compile(wasm);
+    const xdrSections = WebAssembly.Module.customSections(wasmModule, "contractspecv0");
+    if (xdrSections.length === 0) {
+      return Promise.reject(new Error('Could not obtain contract spec from wasm'));
+    }
+    const bufferSection = Buffer.from(xdrSections[0]);
+    const specEntryArray = processSpecEntryStream(bufferSection);
+    const spec = new ContractSpec(specEntryArray);
+    return new ContractClient(spec, options);
+  }
+
+  /**
+   * Generate a ContractClient instance from the contractId and rpcUrl
+   */
+  static async from(options: ContractClientOptions): Promise<ContractClient> {
+    if (!options || !options.rpcUrl || !options.contractId) {
+      return Promise.reject(new TypeError('options must contain rpcUrl and contractId'));
+    }
+    const { rpcUrl, contractId, allowHttp } = options;
+    const serverOpts: Server.Options = { allowHttp: allowHttp?? !rpcUrl.startsWith('https') };
```

Comment:
`options` [has an `allowHttp` field](https://github.com/AhaLabs/js-stellar-sdk/blob/3f134730d015602bf8f964a3c7334c8fb372bff7/src/contract_client/types.ts#L76); we should prefer that instead of inferring it from the `rpcUrl`

Reply:
fixed